hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/NSSummaryTask.java

```diff
@@ -210,10 +267,46 @@ public TaskResult reprocess(OMMetadataManager omMetadataManager) {
           TimeUnit.NANOSECONDS.toMillis(endTime - startTime);
 
       // Log performance metrics
-      LOG.info("Task execution time: {} milliseconds", durationInMillis);
+      LOG.info("NSSummary reprocess execution time: {} milliseconds", durationInMillis);
+      
+      // Reset state to IDLE on successful completion
+      if (success) {
+        REBUILD_STATE.set(RebuildState.IDLE);
+        LOG.info("NSSummary tree reprocess completed successfully with unified control.");
+      }
```

Comment:
State reset only happens on success.
Shouldn't we always reset to either IDLE or FAILED in finally block ?

Reply:
Not sure if I understood here, but we are setting state on both failure and success scenarios. On failure, we are setting FAILED in catch block.